generals/grid.py

```diff
@@ -2,7 +2,71 @@
 from generals.config import PASSABLE, MOUNTAIN
 
 
-class Mapper:
+class Grid:
+    def __init__(self, grid: str):
+        if not Grid.verify_map(grid):
+            raise ValueError("Invalid map layout - generals cannot reach each other.")
+        self._grid_string = grid.strip()
+
+    @property
+    def grid(self):
+        return self._grid_string
+
+    @grid.setter
+    def grid(self, grid: str):
+        grid = grid.strip()
+        if not Grid.verify_map(grid):
+            raise ValueError("Invalid map layout - generals cannot reach each other.")
+        self._grid_string = grid
+
+    @property
+    def numpified_grid(self):
+        return Grid.numpify_grid(self._grid_string)
+
+    @staticmethod
+    def numpify_grid(grid: str) -> np.ndarray:
+        return np.array([list(row) for row in grid.strip().split("\n")])
+
+    @staticmethod
+    def stringify_grid(grid: np.ndarray) -> str:
+        return "\n".join(["".join(row) for row in grid])
+
+    @staticmethod
+    def verify_map(map: str) -> bool:
```

Comment:
Maybe you should rename to `verify_grid` (and also the `map` parameter to `grid`)?